internal/inventory/inventory.go

```diff
@@ -72,13 +80,22 @@ func (a *AssetInventory) Run(ctx context.Context) {
 
 	assetsBuffer := make([]AssetEvent, 0, a.bufferMaxSize)
 	flushTicker := time.NewTicker(a.bufferFlushInterval)
+	fetcherPeriod := time.NewTicker(a.period)
 	for {
 		select {
 		case <-ctx.Done():
 			a.logger.Warnf("Asset Inventory context is done: %v", ctx.Err())
 			a.publish(assetsBuffer)
 			return
 
+		case <-fetcherPeriod.C:
+			a.logger.Debug("starting a new fetch cycle")
+			for _, fetcher := range a.fetchers {
+				go func(fetcher AssetFetcher) {
+					fetcher.Fetch(ctx, a.assetCh)
+				}(fetcher)
+			}
```

Comment:
Shouldn't we remove the the fetcher loop on top of this function?

Reply:
Well, no. I added this so all fetchers run every `fetcherPeriod` tick.

Reply:
And the extra one on top is so that it starts immediately, on second 0. Otherwise it would take until the first `fetcherPeriod` tick to run first fetchers.

Reply:
Hm true, just checked the Ticker and we can't make it run straight away. 

Nitpick: Maybe move this code to a function? Just so we have a nice wrapped function, it prevents if in the future we need to fix something and we do in one place only

Reply:
A note: in manager (used in fetching of cspm mostly), we use a timer, first with a `0` value and then reset with the period value, to do the trick of "trigger now and then per period".

https://github.com/elastic/cloudbeat/blob/b493e0c32173dc4d03da5e1ee56f1b2c9b9625c1/internal/resources/fetching/manager/manager.go#L72-L87

I'm not saying we should do the same with asset inventory since I think the ticker is better for what we want.

Reply:
Thanks. Yeah, I know the trick, but I always find calling `timer.Reset()` every time more messy. Nothing against the implementation, just personal taste I guess.